flow/Histogram.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Histogram.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <flow/Histogram.h>
+#include <flow/Trace.h>
+// TODO: remove dependency on fdbrpc.
+
+// we need to be able to check if we're in simulation so that the histograms are properly
+// scoped to the right "machine".
+// either we pull g_simulator into flow, or flow (and the I/O path) will be unable to log performance
+// metrics.
+#include <fdbrpc/simulator.h>
+
+// pull in some global pointers too:  These types are implemented in fdbrpc/sim2.actor.cpp, which is not available here.  Yuck.
+// If you're not using the simulator, these will remain null, and all should be well.
+
+// TODO: create a execution context abstraction that allows independent flow instances within a process.
```

Comment:
I think  that would be really great to have!! Currently we have bunch of globals instantiated this way, which isn't pretty, but will work as IIRC simulation takes care of switching this when context switching :(

https://github.com/apple/foundationdb/blob/125776491b35b60a1aa03e533faa159351837961/fdbrpc/FlowTransport.actor.cpp#L1503

Reply:
I see.  That doesn't solve the nasty problem I've created here.  `g_network` isn't linked into libflow.a, which means invoking `create_instance()` would create the same linker error as I was trying to avoid.

There are better patterns for accomplishing this, but they're out of scope for this PR.